plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/CondaHostedFacetImpl.java

```diff
@@ -0,0 +1,262 @@
+
+/**
+ * {@link CondaHostedFacetImpl implementation}
+ */
+@Named
+public class CondaHostedFacetImpl
+        extends FacetSupport
+        implements CondaHostedFacet {
+    private final CondaDataAccess condaDataAccess;
+
+    @Inject
+    public CondaHostedFacetImpl(final CondaDataAccess condaDataAccess) {
+        this.condaDataAccess = checkNotNull(condaDataAccess);
+    }
+
+    @Override
+    protected void doInit(Configuration configuration) throws Exception {
+        super.doInit(configuration);
+        getRepository().facet(StorageFacet.class).registerWritePolicySelector(new CondaWritePolicySelector());
+    }
+
+    @Override
+    @TransactionalTouchBlob
+    public Optional<Content> fetch(final String path) {
+        checkNotNull(path);
+
+        StorageTx tx = UnitOfWork.currentTx();
+        Asset maybeAsset = condaDataAccess.findAsset(tx, tx.findBucket(getRepository()), path);
+
+        return Optional.ofNullable(maybeAsset)
+                .map(asset -> {
+                    if (asset.markAsDownloaded()) {
+                        tx.saveAsset(asset);
+                    }
+                    return asset;
+                })
+                .map(asset -> toContent(asset, tx.requireBlob(asset.requireBlobRef())));
+    }
+
+    @Override
+    @TransactionalStoreBlob
+    public Content upload(final String path, final Payload payload) throws IOException {
+        checkNotNull(path);
+        checkNotNull(payload);
+        StorageFacet storageFacet = facet(StorageFacet.class);
+        try (TempBlob tempBlob = storageFacet.createTempBlob(payload, HASH_ALGORITHMS)) {
+            return doPutContent(path, tempBlob, payload);
+        }
+    }
+
+    @Override
+    @TransactionalDeleteBlob
+    public boolean delete(final String path) {
+        checkNotNull(path);
+
+        StorageTx tx = UnitOfWork.currentTx();
+        Bucket bucket = tx.findBucket(getRepository());
+
+        Asset asset = condaDataAccess.findAsset(tx, bucket, path);
+        if (asset == null) {
+            return false;
+        }
+        tx.deleteAsset(asset);
+        return true;
+    }
+
+    protected Content doPutContent(final String path, final TempBlob tempBlob, final Payload payload) throws IOException {
+        StorageTx tx = UnitOfWork.currentTx();
+        CondaPath condaPath = CondaPath.build(path);
+        Asset asset = getOrCreateAsset(condaPath, getRepository());
+
+        AttributesMap contentAttributes = null;
+        if (payload instanceof Content) {
+            contentAttributes = ((Content) payload).getAttributes();
+        }
+
+        Content.applyToAsset(asset, Content.maintainLastModified(asset, contentAttributes));
+
+        AssetBlob assetBlob = tx.setBlob(asset, path, tempBlob, null, payload.getContentType(), false);
+
+        Content content = toContent(asset, assetBlob.getBlob());
+        if (!path.contains("repodata.json")) {
```

Comment:
Maybe make seance create static variable for this?

Reply:
there is one already defined even... -> done